lib/Doctrine/DBAL/Platforms/SQLServer2012Platform.php

```diff
@@ -143,4 +131,35 @@ protected function doModifyLimitQuery($query, $limit, $offset = null)
 
         return $query;
     }
+
+    /**
+     * @param string $query
+     */
+    protected function shouldAddOrderBy($query): bool
```

Comment:
Does this have to be `protected`? Let's make it private unless it's expected to be visible to the subclasses.